include/mbedtls/config.h

```diff
@@ -2996,6 +3037,27 @@
 //#define MBEDTLS_PLATFORM_NV_SEED_READ_MACRO   mbedtls_platform_std_nv_seed_read /**< Default nv_seed_read function to use, can be undefined */
 //#define MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO  mbedtls_platform_std_nv_seed_write /**< Default nv_seed_write function to use, can be undefined */
 
+/**
+ * \brief       This macro is invoked by the library when an invalid parameter
+ *              is detected that is only checked with MBEDTLS_CHECK_PARAMS
+ *              (see the documentation of that option for context).
+ *
+ *              When you leave this undefined here, a default definition is
+ *              provided that invokes the function mbedtls_param_failed(),
+ *              which is declared in platform_util.h for the benefit of the
+ *              library, but that you need to define in your application.
+ *
+ *              When you define this here, this replaces the default
+ *              definition in platform_util.h (which no longer declares the
+ *              function mbedtls_param_failed()) and it is your responsibility
+ *              to make sure this macro expands to something suitable (in
+ *              particular, that all the necessary declarations are visible
+ *              from within the library).
```

Comment:
Question: How can one make sure that the definition of the alternative mbedtls_param_failed() is visible from the library's point of view. As far as I can tell, the library source files expect the declaration to be in platform_util.h, but this is not the case if MBEDTLS_PARAM_FAILED is defined. So, if I define this macro, in what header file should I add the declaration for mbedtls_param_failed()?

Note: In the case of other _ALT macros, the code is arranged such that there is a header with a specific name that is included and compilation must be arranged by the user in such a way that the header is found. However, I do not see a similar arrangement here. Perhaps I am missing something?

Reply:
If you're providing a definition of this macro here, you're already editing `config.h` in a non-trivial way (not supported by `config.pl` for example). So if you need to include some header (eg `assert.h`) or provide a prototype for a function, you may as well do it here while you're at it.

I'm adding a comment suggesting that in the documentation here.